addons/hr_payroll/models/hr_benefit.py

```diff
@@ -279,11 +279,13 @@ def _duplicate_to_calendar_attendance(self):
     def action_validate(self, ids):
         benefits = self.env['hr.benefit'].search([('id', 'in', ids), ('state', '!=', 'validated')])
         benefits.write({'display_warning': False})
-        if not benefits._check_if_error():
-            benefits.write({'state': 'validated'})
-            benefits._duplicate_to_calendar()
-            return True
-        return False
+
+        no_conflict_benefits = benefits.filtered(lambda x: not x._check_if_error())
```

Comment:
No break the batch method. 

Maybe modify ``_check_if_error()`` to return (bool, benefits_in_error) and avoid to use a filtered.